src/coredump/coredump.c

```diff
@@ -1087,40 +1096,67 @@ static int process_socket(int fd) {
                         goto finish;
                 }
 
-                /* The final zero-length datagram carries the file descriptor and tells us
+                /* The final zero-length datagram carries the file descriptors and tells us
                  * that we're done. */
                 if (n == 0) {
                         struct cmsghdr *found;
+                        unsigned n_fds = 0;
+
+                        found = cmsg_find(&mh, SOL_SOCKET, SCM_RIGHTS, CMSG_LEN(sizeof(int)));
```

Comment:
this should be moved into the `if (first)` block below, no need to call this otherwise